session/bootstrap.go

```diff
@@ -1288,6 +1296,76 @@ func upgradeToVer59(s Session, ver int64) {
 	writeOOMAction(s)
 }
 
+func upgradeToVer60(s Session, ver int64) {
+	if ver >= version60 {
+		return
+	}
+	h := &bindinfo.BindHandle{}
+	var err error
+	mustExecute(s, "BEGIN PESSIMISTIC")
+
+	defer func() {
+		if err != nil {
+			mustExecute(s, "ROLLBACK")
+			return
+		}
+
+		mustExecute(s, "COMMIT")
+	}()
+	mustExecute(s, h.LockBindInfoSQL())
+	var recordSets []sqlexec.RecordSet
+	recordSets, err = s.ExecuteInternal(context.Background(), "SELECT original_sql, bind_sql, default_db, charset, collation from mysql.bind_info where source != 'builtin'")
+	if err != nil {
+		debug.PrintStack()
+		logutil.BgLogger().Fatal("upgradeToVer60 error", zap.Error(err))
+	}
+	if len(recordSets) > 0 {
+		defer terror.Call(recordSets[0].Close)
+	}
+	req := recordSets[0].NewChunk()
+	iter := chunk.NewIterator4Chunk(req)
+	p := parser.New()
+	now := types.NewTime(types.FromGoTime(time.Now()), mysql.TypeTimestamp, 3)
+	for {
+		err = recordSets[0].Next(context.TODO(), req)
+		if err != nil {
+			debug.PrintStack()
+			logutil.BgLogger().Fatal("upgradeToVer60 error", zap.Error(err))
+		}
+		if req.NumRows() == 0 {
+			break
+		}
+		updateBindInfo(s, iter, p, now.String())
+	}
+}
+
+func updateBindInfo(s Session, iter *chunk.Iterator4Chunk, p *parser.Parser, now string) {
+	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
+		original := row.GetString(0)
+		bind := row.GetString(1)
+		db := row.GetString(2)
+		charset := row.GetString(3)
+		collation := row.GetString(4)
+		originStmt, err := p.ParseOneStmt(original, charset, collation)
+		if err != nil {
+			debug.PrintStack()
+			logutil.BgLogger().Fatal("updateBindInfo error", zap.Error(err))
+		}
+		bindStmt, err := p.ParseOneStmt(bind, charset, collation)
+		if err != nil {
+			debug.PrintStack()
+			logutil.BgLogger().Fatal("updateBindInfo error", zap.Error(err))
+		}
+		originWithDB := parser.Normalize(utilparser.RestoreWithDefaultDB(originStmt, db))
+		bindWithDB := utilparser.RestoreWithDefaultDB(bindStmt, db)
+		mustExecute(s, fmt.Sprintf(`UPDATE mysql.bind_info SET original_sql=%s, bind_sql=%s, default_db='', update_time=%s where original_sql=%s`,
+			expression.Quote(originWithDB),
+			expression.Quote(bindWithDB),
+			expression.Quote(now),
+			expression.Quote(original)))
+	}
```

Comment:
Shall we drop the `db` column of this table as well?

Reply:
Yes, when using the `create global binding` statement to create a binding, because `db` is added to `origin sql`, the column of `default_db` is empty. The deletion of the column `default_db` is consistent with the above behavior.